src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -1,141 +1,159 @@
-    private final org.slf4j.Logger logger = LoggerFactory.getLogger(OrangebeardTestSystemListener.class);
-    private final String PROPERTY_FILE_NAME = "reportportal.properties";
-    private OrangebeardLogger orangebeardLogger;
-    private ReportPortal reportPortal;
-    private Launch launch;
-    private ToolchainRunningContext context = new ToolchainRunningContext();
+    private String rootPath = getFitnesseRootPath();
     private final OrangebeardTableLogParser htmlChunkParser = new OrangebeardTableLogParser();
-    private Maybe<String> launchId;
+    private OrangebeardLogger orangebeardLogger;
+
+    private boolean local = false;
+
 
+    public OrangebeardTestSystemListener(@Nullable String propertyFile) {
+        if (propertyFile != null) {
+            PROPERTY_FILE_NAME = propertyFile;
+        }
+        rootPath = getFitnesseRootPath();
+    }
 
-    public OrangebeardTestSystemListener(OrangebeardLogger orangebeardLogger, ReportPortal reportPortal, ToolchainRunningContext context) {
-        this.orangebeardLogger = orangebeardLogger;
-        this.reportPortal = reportPortal;
-        this.context = context;
+    public OrangebeardTestSystemListener(@Nullable String propertyFile, String rootPath) {
+        if (propertyFile != null) {
+            PROPERTY_FILE_NAME = propertyFile;
+        }
+        this.rootPath = rootPath;
+    }
+
+    public OrangebeardTestSystemListener(String propertyFile, boolean local) {
+        this(propertyFile);
+        this.local = local;
     }
 
     @Override
     public void testSystemStarted(TestSystem testSystem) {
-        context.setTestSystemName(testSystem.getName());
+        startTestRun(testSystem.getName());
     }
 
-
     @Override
     public void testOutputChunk(String s) {
-        Maybe<String> testId = context.getTestId(context.getLatestTestName());
-        String level = htmlChunkParser.determineLogLevel(s);
+        UUID testId = runContext.getTestId(runContext.getLatestTestName());
 
         if (s.toLowerCase().contains("<table")) {
             s = htmlChunkParser.removeNonTableProlog(s);
             s = htmlChunkParser.applyOrangebeardTableStyling(s);
         }
 
-        String enrichedLog = htmlChunkParser.embedImagesAndStripHyperlinks(s);
-        orangebeardLogger.sendLogData(testId.blockingGet(), launchId.blockingGet(), enrichedLog, level);
-        orangebeardLogger.sendAttachmentsIfPresent(testId.blockingGet(), launchId.blockingGet(), s);
-    }
+        String enrichedLog = htmlChunkParser.embedImagesAndStripHyperlinks(s, rootPath);
 
+        //Workaround for corner case where table contains binary representation with 0x00 unicode chars
+        enrichedLog = enrichedLog.replaceAll("\u0000", "");
+
+        orangebeardLogger.sendLogData(testId, runContext.getTestRun(), enrichedLog, htmlChunkParser.determineLogLevel(s));
+        orangebeardLogger.attachFilesIfPresent(testId, runContext.getTestRun(), s);
```

Comment:
ik zou de if present hier checken, niet in die methode. Dat is een wat beter patroon imo.

Reply:
dan ga je 2x een regex uitvoeren (want hij moet over matches gaan heen lopen) ipv 1x. had expres het al dan niet insturen van attachments aan de logger over gelaten en niet aan de listener (die luistert alleen maar naar de voorbijkomende log chunks..)